agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/entity/AIAgentSubgraph.kt

```diff
@@ -206,8 +206,14 @@ public open class AIAgentSubgraph<Input, Output>(
             val resolvedEdge = currentNode.resolveEdgeUnsafe(context, nodeOutput)
 
             if (resolvedEdge == null) {
-                logger.error { formatLog(context, "Agent stuck in node ${currentNode.name}") }
-                throw AIAgentStuckInTheNodeException(currentNode, nodeOutput)
+                // In we are in the finish node, we need to exit, otherwise we stuck in the node
+                if (currentNode == finish) {
+                    currentInput = nodeOutput
+                    break
+                } else {
```

Comment:
[minor]: You can drop `else` construction here. You can only check for a condition in `if` because you whether break or throw:
```
if (currentNode == finish) {
    currentInput = nodeOutput
    break
}

logger.error { formatLog(context, "Agent stuck in node ${currentNode.name}") }
throw AIAgentStuckInTheNodeException(currentNode, nodeOutput)
```
Or check the throw condition instead if you like to check first.